plugins/modules/packaging/os/pacman.py

```diff
@@ -430,8 +474,22 @@ def _install_packages_for_real(pacman_verb, pkglist):
         if pkgs_to_install_from_url:
             _install_packages_for_real("--upgrade", pkgs_to_install_from_url)
 
-        self.exit_params["packages"] = installed_pkgs
-        self.add_exit_infos("Installed %d package(s)" % len(installed_pkgs))
+        # set reason
+        if pkgs_to_set_reason:
+            cmd = [self.pacman_path, "--noconfirm", "--database"]
+            if self.m.params["reason"] == "dependency":
+                cmd.append("--asdeps")
+            else:
+                cmd.append("--asexplicit")
+            cmd.extend(pkgs_to_set_reason)
+
+            rc, stdout, stderr = self.m.run_command(cmd, check_rc=False)
```

Comment:
Why not have
```suggestion
            rc, stdout, stderr = self.m.run_command(cmd, check_rc=True)
```
and remove the explicit `rc` check below?

Reply:
I don't know. The same code is used in various other places:
-https://github.com/Minei3oat/community.general/blob/pacman_install_reason/plugins/modules/packaging/os/pacman.py#L403
- https://github.com/Minei3oat/community.general/blob/pacman_install_reason/plugins/modules/packaging/os/pacman.py#L467
- and multiple other in the other operations

Reply:
I guess all these places should be changed then. But we can also do that later.